consul: Remove implicit workload identity when task has a template. (#25298)
hashicorp/nomad: website/content/docs/upgrade/upgrade-specific.mdx

Refer to [Migrating to Using Workload Identity with Vault][] for more
details.

#### Consul template implicit workload identity removal

> Nomad will no longer create an implicit Consul identity for workloads that don't

Reviewer: ```suggestion
Nomad no longer creates an implicit Consul identity for workloads that don't
```

Not an expert here but I think our guidelines say we should use present tense in phrases like that.

register services with Consul. Jobs that require Consul tokens for template

> rendering will be required to provide an identity.

Reviewer 2: ```suggestion
rendering must provide an identity.
```
style: present tense

Tasks that include template blocks utilizing Consul API functionality such as

> `key`, `node`, and `service` will now need to specify an [`identity`][] block,

Reviewer 3: We should still be creating the identity block implicitly when there's a `consul` block present.

> `key`, `node`, and `service` will now need to specify an [`identity`][] block,

Reviewer 2: ```suggestion
`key`, `node`, and `service` must now specify an [`identity`][] block,
```
style: present tense
